packages/core/src/application.ts

```diff
@@ -88,6 +91,9 @@ export class Application extends Context {
     response: ServerResponse,
   ) {
     this._setupHandlerIfNeeded();
+    if (request.method === 'GET' && request.url === '/openapi.json') {
+      return this._serveOpenApiSpec(request, response);
```

Comment:
I would add a comment above the code that gives the 1st reason for why it's done this way, just to prevent someone from "optimizing" it away in future.